docs/fp_primer.md

```diff
@@ -0,0 +1,495 @@
+accuracies. 
+
+For examples, `tan(x) inherits from sin(x)/cos(x)`, one can take the defined
+rules and manually build up a bespoke solution for checking the results, but
+this is tedious, error-prone, and doesn't allow for code re-use.
+
+Instead, it would be better if there was a single conceptual framework that one
+can express all the 'simple' accuracy requirements in, and then have a mechanism
+for composing them to define inherited accuracies.
+
+In the WebGPU CTS this is done via the concept of acceptance intervals, which is
+derived from a similar concept in the Vulkan CTS, though implemented
+significantly differently.
+
+The core of this idea is that each of different accuracy types can be integrated
+into the definition of the operation, so that instead of transforming an input
+from the domain to a point in the range, the operation is producing an interval
+in the range, that is the acceptable values an implementation may emit.
+
+
+The simple accuracies can be defined as follows:
+
+1. *Exact*
+   
+   `f(x) => [X, X]`
+
+
+2. *Correctly Rounded*
+   
+   If `X` is precisely defined as a f32
+
+   `f(x) => [X, X]`
+
+   otherwise,
+
+   `[a, b]` where `a` is `min(X - a)`, and `b` is `min(b - X)`
+
+
+3. *Absolute Error*
+
+   `f(x) => [ X - ε, X + ε ]`, where ε is the absolute error value
+
+
+4. **ULP Error**
+
+   `f(x) = X => [X - n*ULP(X), X + n*ULP(X)]`
+
+As defined, these definitions handle mapping from a point in the domain into an
+interval in the range.
+
+This is insufficient for implementing inherited accuracies, since inheritance
+sometimes involve mapping domain intervals to range intervals, i.e. 
+`f([a, b]) = [A, B]`.
+
+Given that floating point numbers have a finite number of precise values for any
+given interval, one could implement just running the accuracy computation for
+every point in the interval and then spanning together the resultant intervals.
+That would be very inefficient though and make your reviewer sad to read.
+
+For mapping intervals to intervals the key insight is that we only need to be
+concerned with the extremas of the operation in the interval, since the
+acceptance interval is the bounds of the possible outputs.
+
+In more precise terms:
+```
+  f(x) => X, x = [a, b] and X = [A, B]
+  
+  X = [min(f(x)), max(f(x))]
+  X = [min(f([a, b])), max(f([a, b]))]
+  X = [f(m), f(M)]
+```
+where m and M are in `[a, b]`, `m <= M`, and produce the min and max results
+respectively.
+
+So how do we find the minima and maxima for our operation in the domain?
+
+The general solution for this requires using calculus to calculate the
+derivative of `f`, `f'`, and then find the zeroes `f'` to find inflection 
+points of `f`.
+
+Thankfully we do not need a general solution for the CTS, since all the builtin
+operations are defined in the spec, so `f` is from a known set of options.
+
+These operations can be divided into two broad categories: monotonic, and 
+non-monotonic.
```

Comment:
this is always with respect to an interval (which becomes important later, I think).

So

> ... monotonic, and non-monotonic, with respect to an interval.

Reply:
Done